### ISSUE TITLE
broken /who flags

### ISSUE THREAD

**Issue by TurkDesk** [inspircd/inspircd#656]
the flag i to search idents broken when used with flag h
you can do /who Me i
## you can not do /who Me ih

the flag r to search gecos is broken, it also catchs nicks
if there is nick "MeAndI" the command /who Me\* r
lists the nick MeAndI as well

**Comment by TurkDesk** [comment]
oper check seems fails for +s and/or +p chans in /who where an oper could see these chans by /whois so these users should not be listed by /who as in no channels

**Comment by Shawn-Smith** [comment]
@TurkDesk That issue is unrelated to broken /who flags.

That issue is here: https://github.com/inspircd/inspircd/blob/master/src/coremods/core_who.cpp#L43

````
if (!memb->chan->IsModeSet(secretmode))
````
Would need to be something like
````
if (user->HasPrivPermission("users/auspex") || !memb->chan->IsModeSet(secretmode))
````

**Comment by SaberUK** [comment]
>/who #channel o shouldn't show the senders nickname if the sender isn't an oper.

Fixed in 11e67c494a9a1e7796c7b6ec0a60c88019181020.

>/who * o and /who #channel o should display visible opers, rather than only invisible ones.

This seems to work as intended from what I can see?

---

If you encounter any more WHO issues then please open a new issue.
